### ISSUE TITLE
Theano to Tensorflow backend with correct dim ordering

### ISSUE THREAD

**Issue by rcmalli** [fchollet/keras#3994]
Current solution in the [wiki page](https://github.com/fchollet/keras/wiki/Converting-convolution-kernels-from-Theano-to-TensorFlow-and-vice-versa) for switching the backend for pretrained models is limited to fixed dimension ordering. For an Example, If you trained a model with Theano backend and used 'th' ordering, you have to preserve the dimension ordering as 'th' when you switch to Tensorflow. What is the efficient and correct way to convert model fully to another backend with correct ordering ? 

Example:   dim_order = 'th' backend = theano  --> dim_order = 'tf' backend = tensorflow

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by rcmalli** [comment]
Update 1 : I am closing this issue until there is a better solution for it. Here is the steps I followed to convert fully a pretrained model.

1 ) Adjust your ~/.keras/keras.json  dim = 'th' backend  = 'tensorflow'
2) load your model and convert it as suggested in [wiki link](https://github.com/fchollet/keras/wiki/Converting-convolution-kernels-from-Theano-to-TensorFlow-and-vice-versa)
3) iterate over your layers  and get transpose of the Conv layers with `numpy.transpose()` and save them.
4) Adjust your ~/.keras/keras.json  dim = 'tf' backend  = 'tensorflow'
5) Load your model with empty weights
6) Iterate over your layers and fill with your transposed weights using `model.get_layer(layer.name).set_weights(weight)` function
7) save your model with `model.save_weights()`

**Comment by titu1994** [comment]
@rcmalli This solution works if you only have Convolutional layers and no Dense layer for the final classification. 

According to @fchollet (in the [google group answer](https://groups.google.com/forum/#!topic/keras-users/E1W4HpuxxFw)) : 
```
Besides converting (transposing) the kernels, you need to:
- dimshuffle the kernels
- shuffle the rows of the weights of the Dense layer you put after your conv layers, to match the dimshuffle you just did

That's it. That's how Keras applications were generated (in particular the VGG16 and VGG19 models).
```

However, I can't think of a proper way to actually perform the second step (shuffle the rows of the first dense layer after the last conv layer). 

@fchollet, if possible, could you provide a code snippet to demonstrate ? If not, could you provide a gist of the script you used to convert the VGG 16 / 19 weights (from theano to tensorflow or vice-versa) ?
